src/rust/receiver-mock/src/router/otlp.rs

```diff
@@ -103,6 +88,55 @@ fn get_otlp_lines_from_logs(resource_logs: &logsv1::ResourceLogs) -> Vec<String>
         .collect()
 }
 
+pub async fn handler_receiver_otlp_metrics(
+    req: HttpRequest,
+    body: web::Bytes,
+    app_state: web::Data<AppState>,
+    opts: web::Data<options::Options>,
+) -> impl Responder {
+    let remote_address = get_address(&req);
+
+    let content_type = get_content_type(&req);
+
+    if let Some(response) = try_dropping_data(&opts, &content_type) {
+        return response;
+    }
+
+    match content_type.as_str() {
+        OTLP_PROTOBUF_FORMAT_CONTENT_TYPE => {
+            let metrics_data: metricsv1::MetricsData = match metricsv1::MetricsData::decode(&mut Cursor::new(body)) {
+                Ok(data) => data,
+                Err(_) => return HttpResponse::BadRequest().body("Unable to parse body"),
+            };
+            let mut result = MetricsHandleResult::new();
+            for resource_metrics in metrics_data.resource_metrics {
+                for instrumentation_lib_metrics in resource_metrics.instrumentation_library_metrics {
+                    for metric in instrumentation_lib_metrics.metrics {
+                        let _metric_str = otlp_metric_to_prometheus_string(&metric);
+                        if opts.print.metrics {
+                            println!("metrics => {} {}", metric.name, metric.description);
+                        }
+
+                        result.handle_metric(metric.name);
+                        result.handle_ip(remote_address);
+                    }
+                }
+            }
+
+            app_state.add_metrics_result(result, &opts);
+        }
+        &_ => {
+            return get_invalid_header_response(&content_type);
+        }
+    }
+
+    HttpResponse::Ok().body("")
+}
+
+fn otlp_metric_to_prometheus_string(metric: &metricsv1::Metric) -> String {
+    format!("{}{{{}}} {} {}", metric.name, metric.description, "".to_string(), "".to_string())
+}
+
```

Comment:
I have problems with understanding how this should be done. `Metric` type contains fields `name`, `description`, `data` and `unit`. Would all the labels be the same for every metric in a request, extracted from `MetricsData` like in case of logs? How to get the timestamp?